api/src/main/java/org/pmiops/workbench/actionaudit/adapters/AuthDomainAuditAdapterImpl.java

```diff
@@ -0,0 +1,54 @@
+package org.pmiops.workbench.actionaudit.adapters;
+
+import java.time.Clock;
+import javax.inject.Provider;
+import org.pmiops.workbench.actionaudit.ActionAuditEvent;
+import org.pmiops.workbench.actionaudit.ActionAuditService;
+import org.pmiops.workbench.actionaudit.ActionType;
+import org.pmiops.workbench.actionaudit.AgentType;
+import org.pmiops.workbench.actionaudit.TargetType;
+import org.pmiops.workbench.actionaudit.targetproperties.AccountTargetProperty;
+import org.pmiops.workbench.db.model.DbUser;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Qualifier;
+import org.springframework.stereotype.Service;
+
+@Service
+public class AuthDomainAuditAdapterImpl implements AuthDomainAuditAdapter {
+
+  private ActionAuditService actionAuditService;
+  private Clock clock;
+  private Provider<DbUser> adminDbUserProvider;
+  private Provider<String> actionIdProvider;
+
+  @Autowired
+  public AuthDomainAuditAdapterImpl(
+      ActionAuditService actionAuditService,
+      Clock clock,
+      Provider<DbUser> adminDbUserProvider,
```

Comment:
what makes this an `admin` user provider? I'm assuming the provider being given is the same as the other ones since there isn't a qualifier.

Reply:
It's just to remind me which user is which, since in this case the administrative user is the agent and the main user is the target. It's not explicitly guaranteed that the user coming out of that provider has an admin authority, but if I understand the code I'm instrumenting, that should be the case if we land in one of these methods.